### PR TITLE
mysql: add ddl-related error code and name

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
Some errors in TiDB do not have an error code. Add parts of them.

Related TiDB PR: https://github.com/pingcap/tidb/pull/13329

Side effects

 - Breaking backward compatibility

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @jackysp

File: mysql/errname.go [hunk 1]
```diff
@@ -968,6 +968,21 @@ var MySQLErrName = map[uint16]string{
 	ErrColumnStateNonPublic:       "can not use non-public column",
 	ErrIndexStateCantNone:         "index %s can't be in none state",
 	ErrInvalidRecordKey:           "invalid record key",
+	ErrInvalidDDLWorker:           "Invalid DDL worker",
+	ErrUnsupportedDDLOperation:    "Unsupported",
```
Comment (winkyao) [inline mysql/errname.go]: Unsupported DDL Opearation?

File: mysql/errname.go [hunk 2]
```diff
@@ -968,6 +968,21 @@ var MySQLErrName = map[uint16]string{
...
+	ErrNotOwner:                   "TiDB server is not owner",
```
Comment (winkyao) [inline mysql/errname.go]: TiDB server is not a DDL owner

File: mysql/errname.go (hunk 1 above)
Comment (AilinKid) [inline mysql/errname.go]: Maybe this name is too short? Unsupported what？

File: mysql/errname.go [hunk 3]
```diff
@@ -968,6 +968,21 @@ var MySQLErrName = map[uint16]string{
...
+	ErrCantDecodeIndex:            "Cannot decode index value, because %s",
+	ErrInvalidDDLJob:              "Invalid DDL job",
+	ErrInvalidJobFlag:             "Invalid job flag",
```
Comment (winkyao) [inline mysql/errname.go]: Invalid DDL job flag

File: mysql/errname.go [hunk 4]
```diff
@@ -968,6 +968,21 @@ var MySQLErrName = map[uint16]string{
...
+	ErrWaitReorgTimeout:           "Timeout waiting for data reorganization",
+	ErrInvalidStoreVersion:        "Invalid storage current version: %d",
+	ErrUnknownTypeLength:          "Unknown length for type %d",
+	ErrUnknownFractionLength:      "Unknown length for type %d and fraction %d",
+	ErrInvalidJobVersion:          "Version %d of DDL job is greater than current (current what?) %d",
+	ErrInvalidSplitRegionRanges:   "Failed to split region ranges",
+	ErrReorgPanic:                 "Reorg worker panic",
+	ErrInvalidDDLState:            "Invalid %s state",
+	ErrCancelledDDLJob:            "Cancelled DDL job",
```
Comment (AilinKid) [inline mysql/errname.go]: Maybe `Cancelled DDL job error` or `Cancelled DDL job failed`? `Cancelled DDL job` seems like a derogatory term, the job cancelled successfully.

File: mysql/errname.go [hunk 5]
```diff
@@ -968,6 +968,21 @@ var MySQLErrName = map[uint16]string{
...
+	ErrInvalidDDLJob:              "Invalid DDL job",
+	ErrInvalidJobFlag:             "Invalid job flag",
+	ErrWaitReorgTimeout:           "Timeout waiting for data reorganization",
+	ErrInvalidStoreVersion:        "Invalid storage current version: %d",
+	ErrUnknownTypeLength:          "Unknown length for type %d",
+	ErrUnknownFractionLength:      "Unknown length for type %d and fraction %d",
+	ErrInvalidJobVersion:          "Version %d of DDL job is greater than current (current what?) %d",
+	ErrInvalidSplitRegionRanges:   "Failed to split region ranges",
+	ErrReorgPanic:                 "Reorg worker panic",
+	ErrInvalidDDLState:            "Invalid %s state",
```
Comment (zimulala) [inline mysql/errname.go]:
```suggestion
	ErrInvalidDDLState:            "Invalid %s state %s",
```
e.g. `Invalid table state write only`.

File: mysql/errname.go (hunk 1 above)
Comment (bb7133) [reply mysql/errname.go]: This is the common prefix of all UnsupportedDDLOperation errors, for example:

```
errCantDropColWithIndex     = terror.ClassDDL.New(mysql.ErrUnsupportedDDLOperation, mysql.MySQLErrName[mysql.ErrUnsupportedDDLOperation]+" drop column with index")
```

I will add a placeholder at the end to make it read-friendly.

Comment (bb7133) [reply mysql/errname.go]: Please check the comment above.

File: mysql/errname.go (hunk 4 above)
Comment (bb7133) [reply mysql/errname.go]: This error DOES mean "DDL job is canceled successfully"... Maybe we should re-consider this message, but it's better to make it in another PR.

File: mysql/errname.go (hunk 3 above)
Comment (zimulala) [reply mysql/errname.go]: Please address this comment.

Comment (bb7133) [reply mysql/errname.go]: Done, thanks

**Pull request merged**